src/components/common/TokenImg.tsx

```diff
@@ -46,23 +52,36 @@ const tokensIconsFilesByAddress: Record<string, string> = tokensIconsRequire.key
   return acc
 }, {})
 
-export const TokenImg: React.FC<Props> = (props) => {
-  const { address, addressMainnet, symbol, name } = props
+type ImageLoadProps = Pick<React.ImgHTMLAttributes<HTMLImageElement>, 'src' | 'alt' | 'onError'>
 
-  let iconFile = tokensIconsFilesByAddress[address.toLowerCase()]
-  if (!iconFile && addressMainnet) {
-    iconFile = tokensIconsFilesByAddress[addressMainnet.toLowerCase()]
-  }
+const useFailOnceImage = ({ address, addressMainnet, symbol, name }: Omit<Props, 'faded'>): ImageLoadProps => {
+  return useMemo(() => {
+    let iconFile = tokensIconsFilesByAddress[address.toLowerCase()]
+    if (!iconFile && addressMainnet) {
+      iconFile = tokensIconsFilesByAddress[addressMainnet.toLowerCase()]
+    }
+
+    const iconFileUrl: string | undefined = iconFile
+      ? tokensIconsRequire(iconFile).default
+      : getImageUrl(addressMainnet || address)
 
-  const iconFileUrl: string | undefined = iconFile
-    ? tokensIconsRequire(iconFile).default
-    : getImageUrl(addressMainnet || address)
+    // if we know the image failed before, use fallback image right away
+    const imgSrc = iconFileUrl && !failedTokenImages.has(iconFileUrl) ? iconFileUrl : unknownTokenImg
+
+    // TODO: Simplify safeTokenName signature, it doesn't need the addressMainnet or id!
+    // https://github.com/gnosis/dex-react/issues/1442
+    const safeName = safeTokenName({ address, symbol, name })
+
+    return { src: imgSrc, alt: safeName, onError: _loadFallbackTokenImage }
+  }, [address, addressMainnet, symbol, name])
+}
+
+export const TokenImg: React.FC<Props> = (props) => {
+  const { faded } = props
 
-  // TODO: Simplify safeTokenName signature, it doesn't need the addressMainnet or id!
-  // https://github.com/gnosis/dex-react/issues/1442
-  const safeName = safeTokenName({ address, symbol, name })
+  const imageProps = useFailOnceImage(props)
 
-  return <Wrapper alt={safeName} src={iconFileUrl} onError={_loadFallbackTokenImage} {...props} />
+  return <Wrapper {...imageProps} faded={faded} />
```

Comment:
Just to learn your stance here: Why is it better to pass an object with props here, than passing them as separate props? I'd be inclined to say, viewing them explicitly separated reads quicker. On the other hand, passing a single props object, arguably is easier to read too, given logic is handled separately in the `useFailOnceImage` function.

Reply:
it's not better or worse, just different
Now passing all `props` was bad, because `{address, symbol, name, ...}` ended up on the `Wrapper`, and while most of them got filtered by react, `name` gets applied, which doesn't break anything,but just not good. Also extra unnecessary props can cause extra unnecessary rerenders when they change
`imageProps` on the other hand are constructed as `Pick<ImgHTMLAttributes, **>` so they are real img attributes